File: prog/ruby/printheadlines.rb
require 'rss'
require 'time'

feed = RSS::Parser.parse("https://www.heise.de/newsticker/"\
       "heise-atom.xml")
feed.items.each do |item|
    d = Time.parse(item.updated.content.to_s)
    puts "* [#{d.strftime('%Y-%m-%d %H:%M:%S')}]:"\
    "#{item.title.content}"
end

File: prog/python/printheadlines.py
#!/usr/bin/env python3

import feedparser
import time

feed = feedparser.parse("https://www.heise.de/newsticker/heise-atom.xml")

[ print("* [%s]: %s" %
    (time.strftime("%Y-%m-%d %H:%M:%S", entry.published_parsed),
     entry.title))
    for entry in feed.entries ]
